src/com/palmergames/bukkit/towny/TownyMessaging.java

```diff
@@ -813,6 +814,50 @@ public static void sendJailList(Player player, Town town, int page, int total) {
 		audience.sendMessage(pageFooter);
 	}
 	
+	public static void sendPlotGroupList(CommandSender sender, Town town, int page, int total) {
+		int groupCount = town.getPlotGroups().size();
+		int iMax = Math.min(page * 10,  groupCount);
+		List<PlotGroup> groups = new ArrayList<>(town.getPlotGroups());
+		
+		TextComponent[] groupsFormatted;
+		if ((page * 10) > groupCount) {
+			groupsFormatted = new TextComponent[groupCount % 10];
+		} else {
+			groupsFormatted = new TextComponent[10];
+		}

```

Comment:
Instead of a textcomponent array, this could probably just be one textcomponent that gets appended each time and sent to the player in one go.

Reply:
I've found that the appending can be a bit messy, some times the colouring will be applied to the wrong component based on where the )'s are. I actually fixed one instance of it in the map hovering while adding/testing the group to the ascii map hovering.